containers/servlet/src/main/java/org/glassfish/tyrus/servlet/TyrusServletContainerInitializer.java

```diff
@@ -83,34 +83,9 @@ public void onStartup(Set<Class<?>> classes, final ServletContext ctx) throws Se
                                     DebugContext.TracingThreshold.TRACE);
 
         final ApplicationEventListener applicationEventListener = createApplicationEventListener(ctx);
-        final TyrusServerContainer serverContainer = new TyrusServerContainer(classes) {
-
-            private final WebSocketEngine engine =
-                    TyrusWebSocketEngine.builder(this)
-                                        .applicationEventListener(applicationEventListener)
-                                        .incomingBufferSize(incomingBufferSize)
-                                        .maxSessionsPerApp(maxSessionsPerApp)
-                                        .maxSessionsPerRemoteAddr(maxSessionsPerRemoteAddr)
-                                        .parallelBroadcastEnabled(parallelBroadcastEnabled)
-                                        .tracingType(tracingType)
-                                        .tracingThreshold(tracingThreshold)
-                                        .build();
-
-            @Override
-            public void register(Class<?> endpointClass) throws DeploymentException {
-                engine.register(endpointClass, ctx.getContextPath());
-            }
-
-            @Override
-            public void register(ServerEndpointConfig serverEndpointConfig) throws DeploymentException {
-                engine.register(serverEndpointConfig, ctx.getContextPath());
-            }
-
-            @Override
-            public WebSocketEngine getWebSocketEngine() {
-                return engine;
-            }
-        };
+        final TyrusServerContainer serverContainer = new TyrusServerContainerImpl(classes, applicationEventListener,
```

Comment:
Converted anonymous inner class to static member class, so context isn't implicitly brought in and leaked